### PR TITLE
add support for JMS

### PR DESCRIPTION
**Pull request by burtbeckwith**

### CODE REVIEW

**Comment by wetted** [comment]
This should probably target 2.4.x once that branch is available.

File: starter-core/src/main/java/io/micronaut/starter/feature/messaging/jms/AbstractJmsFeature.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+public abstract class AbstractJmsFeature implements JmsFeature {
+
+    @Override
+    public boolean supports(ApplicationType applicationType) {
+        return true;
+    }
+
+    @Override
+    public String getMicronautDocumentation() {
+        return "https://micronaut-projects.github.io/micronaut-jms/latest/guide/index.html";
```
Comment (wetted) [inline starter-core/src/main/java/io/micronaut/starter/feature/messaging/jms/AbstractJmsFeature.java:29]: This link is unreachable and results in a 404 error.

**Review by wetted — APPROVED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/messaging/jms/AbstractJmsFeature.java (hunk 1 above)
Comment (alvarosanchez) [reply starter-core/src/main/java/io/micronaut/starter/feature/messaging/jms/AbstractJmsFeature.java:29]: It should point to the snapshot URL. Also, the feature should be marked as preview

**Review by alvarosanchez — CHANGES_REQUESTED** [review]

**Review by alvarosanchez — APPROVED** [review]

File: starter-core/src/test/groovy/io/micronaut/starter/feature/messaging/jms/JmsSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+class JmsSpec extends BeanContextSpec implements CommandOutputFixture {
+
+    void 'test README.md with feature jms contains links to micronaut docs'() {
+        when:
+        def output = generate([feature])
+        def readme = output['README.md']
+
+        then:
+        readme
+        readme.contains 'https://micronaut-projects.github.io/micronaut-jms/latest/guide/index.html'
```
Comment (alvarosanchez) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/messaging/jms/JmsSpec.groovy:22]: This should be snapshot too

**Review by alvarosanchez — CHANGES_REQUESTED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/messaging/jms/AbstractJmsFeature.java [hunk 3]
```diff
@@ -26,6 +26,11 @@ public boolean supports(ApplicationType applicationType) {
 
     @Override
     public String getMicronautDocumentation() {
-        return "https://micronaut-projects.github.io/micronaut-jms/latest/guide/index.html";
+        return "https://micronaut-projects.github.io/micronaut-jms/snapshot/guide/index.html";
```
Comment (graemerocher) [inline starter-core/src/main/java/io/micronaut/starter/feature/messaging/jms/AbstractJmsFeature.java:29]: test needs updating

File: test-core/src/test/groovy/io/micronaut/starter/core/test/create/CreateMessagingSpec.groovy [hunk 4]
```diff
@@ -38,7 +37,54 @@ class CreateMessagingSpec extends CommandSpec {
         where:
         [lang, buildTool, feature] << LanguageBuildCombinations.combinations(
                 beanContext.streamOfType(MessagingFeature.class)
+                        .filter({ f -> f.name != 'jms-sqs'} )
```
Comment (burtbeckwith) [inline test-core/src/test/groovy/io/micronaut/starter/core/test/create/CreateMessagingSpec.groovy:40]: I'll add a default instance of AmazonSQS in M2 and update the docs to explain that it must be overridden in client apps, and revert these test changes with that

**Review by alvarosanchez — APPROVED** [review]

**Pull request merged**